test/lib/s3_test.exs

```diff
@@ -337,6 +338,28 @@ defmodule ExAws.S3Test do
              ])
   end
 
+  test "#delete_multiple_objects sha256" do
+    set_content_hash_algorithm()
+
+    expected = %Operation.S3{
+      body:
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Delete><Object><Key>foo</Key></Object><Object><Key>bar</Key><VersionId>v1</VersionId></Object><Object><Key>special characters: &apos;&quot;&amp;&lt;&gt;&#13;&#10;</Key></Object></Delete>",
+      bucket: "bucket",
+      path: "/?delete",
+      headers: %{"x-amz-checksum-sha256" => "Hsce+MZp64Uy8CwyResJ3y13YGw6jDAAdVGFmFPKPSs="},
+      http_method: :post
+    }
+
+    assert expected ==
+             S3.delete_multiple_objects("bucket", [
+               "foo",
+               {"bar", "v1"},
+               "special characters: '\"&<>\r\n"
+             ])
+
+    unset_content_hash_algorithm()
```

Comment:
I think putting the `unset` call in [on_exit/2](https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#on_exit/2) will provide a stronger guarantee the application env gets restored (or deleted)